siege_utilities/geo/s2_utils.py

```
@@ -0,0 +1,676 @@
+    (Python int, numpy.int64, numpy.uint64) for round-trip convenience.
+    """
+    _require_s2()
+    if isinstance(cell_or_token, str):
+        return _s2.CellId.from_token(cell_or_token).id()
+    try:
+        return int(cell_or_token)
+    except (TypeError, ValueError):
+        raise TypeError(
+            f"Expected int64 cell id or hex token string, got "
+            f"{type(cell_or_token).__name__}"
+        )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)